js_modules/dagit/packages/core/src/assets/usePartitionHealthData.tsx

```diff
@@ -49,78 +49,101 @@ export type PartitionDimensionSelection = {
 };
 
 export function buildPartitionHealthData(data: PartitionHealthQuery, loadKey: AssetKey) {
-  const dimensions =
+  const __dims =
     data.assetNodeOrError.__typename === 'AssetNode'
       ? data.assetNodeOrError.partitionKeysByDimension
       : [];
 
-  const materializationStatus = (data.assetNodeOrError.__typename === 'AssetNode' &&
-    data.assetNodeOrError.partitionMaterializationStatus) || {
-    __typename: 'MaterializationStatusSingleDimension',
-    materializationStatus: [],
+  const materializedPartitions = (data.assetNodeOrError.__typename === 'AssetNode' &&
+    data.assetNodeOrError.materializedPartitions) || {
+    __typename: 'DefaultPartitions',
+    unmaterializedPartitions: [],
+    materializedPartitions: [],
   };
 
-  const stateByKey = Object.fromEntries(
-    materializationStatus.__typename === 'MaterializationStatusSingleDimension'
-      ? materializationStatus.materializationStatus.map((materialized, idx) => [
-          dimensions[0].partitionKeys[idx],
-          materialized ? PartitionState.SUCCESS : PartitionState.MISSING,
-        ])
-      : materializationStatus.materializationStatusGrouped.map((dim0, idx0) => [
-          dimensions[0].partitionKeys[idx0],
-          Object.fromEntries(
-            dim0.map((materialized, idx1) => [
-              dimensions[1].partitionKeys[idx1],
-              materialized ? PartitionState.SUCCESS : PartitionState.MISSING,
-            ]),
-          ),
-        ]),
-  );
+  // The backend re-orders the dimensions only for the materializedPartitions ranges so that
+  // the time partition is the "primary" one, even if it's dimension[1] elsewhere.
+  // This matches the way we display them in the UI and makes some common data retrieval faster,
+  // but Dagit's internals always use the REAL ordering of the partition keys, we need to flip
+  // everything in this function to match the range data.
+  const isRangeDataInverted =
+    materializedPartitions.__typename === 'MultiPartitions' &&
+    materializedPartitions.primaryDimensionName !== __dims[0].name;
```

Comment:
`__dims[0].name` looks a little risky for a type error, since it looks like there is a case where `__dims` can be an empty array.

Reply:
Hmm I think you're dwefinitely right about these index accesses - I'll add a few more defensive checks in here.